src/Oscoin/Storage/Block/Orphanage.hs

```diff
@@ -515,11 +512,22 @@ insertOrphan b (storeOrphan b -> o) =
                    $ tips orph
             }
 
+-- | A variation over 'Vector.!' that fails with a stacktrace.
+unsafeIndex :: HasCallStack => Vector a -> Int -> a
+v `unsafeIndex` ix = withFrozenCallStack $
+    let msg =  "Orphanage.unsafeIndex: index out of range ("
+            ++ show ix
+            ++ ","
+            ++ show (length v)
+            ++ ")"
+    in fromMaybe (Prelude.error msg) (v Vector.!? ix)
+
 findCandidate :: (ChainCandidate c s -> Bool)
               -> Candidates c s
               -> Maybe (ChainCandidate c s, Index)
 findCandidate predFn xs =
-    case List.filter (\(c,_) -> predFn c) (zip xs [0..]) of
+    let zipped = Vector.zip xs (Vector.fromList [0..])
+    in case Vector.toList $ Vector.filter (\(c,_) -> predFn c) zipped of
```

Comment:
nit: the `protolude` `head` should work here without converting to a list

Reply:
I do wonder why in the previous incarnation of this function I did request (at the pattern matching level) that a singleton list has to be returned. I don't think that's necessary, so indeed I think we can use `Protolude.head` here.

Reply:
be0b7391